stablehlo/dialect/TypeInference.cpp

```diff
@@ -1388,8 +1388,9 @@ LogicalResult inferDynamicUpdateSliceOp(
 
   // (C1)
   if (operandType.hasRank()) {
-    inferredReturnShapes.emplace_back(operandType.getShape(),
-                                      operandType.getElementType());
+    inferredReturnShapes.emplace_back(
+        operandType.getShape(), operandType.getElementType(),
+        operandType.cast<RankedTensorType>().getEncoding());
```

Comment:
What are our plans for updating the rest of the code? More specifically, what's going to happen to C6?

Reply:
I think this leads to a more general question: what do we do for an existing "X <= Y" check where X is a dim from one tensor, and Y is a dim from a different tensor, in the new bounds scenario? 
I would propose a util function below to replace:
```
assert(X <= Y);
```
with 
```
assert(LessEqualThanWithBounds(X, Y, A, B);
```
```
//  c4:  ?              any             true
//  c5:  any            ?               true
//  c0:  X              Y               assert X <= Y
//  c2:  X              ?, B            assert X <= B
//  c2:  ?, A           Y               true(?)
//  c6:  ?, B           ?, C            true(?)
LogicalResult LessEqualThanWithBounds(
    Optional<Location> location, int64_t dim, 
    int64_t leftSize, int64_t rightSize, 
    int64_t leftBound, int64_t rightBound) {
  // TODO
}
```

Reply:
Thanks for summarizing. 
Added a ticket  to track this.